hypothesis-python/src/hypothesis/internal/conjecture/minimizer.py

```diff
@@ -271,12 +291,14 @@ def shift_right(mid):
             first = False
             change_counter = self.changes
 
-            self.sort()
+            sorted = self.sort()
```

Comment:
I'm not sure checking if the sort failed is useful: The `partial_sort` pass is a fairly cheap check for sortedness anyway, and the intervening passes could cause it to be unsorted.